packages/clerk-js/src/ui/hooks/usePagination.ts

```diff
@@ -0,0 +1,14 @@
+import { useCallback, useState } from 'react';
+
+type UsePaginationProps = {
+  defaultPage: number;
+};
+
+export const usePagination = (props: UsePaginationProps = { defaultPage: 1 }): [number, (p: number) => void] => {
+  const { defaultPage } = props;
+  const [page, setPage] = useState(defaultPage);
+
+  const changePage = useCallback((p: number) => setPage(p), [setPage]);
+
+  return [page, changePage];
+};
```

Comment:
Appreciate the feedback! Not sure this works as expected.
```
export const usePagination = (props: UsePaginationProps) => {
  const { defaultPage } = props || { defaultPage: 1 };
  ...
};
```
I imagine it's better to declare each variable separately in the same manner as you did here, in order to avoid clunkiness:
```
export const usePagination = (props: UsePaginationProps) => {
  const [ page, changePage ] = useState(props.defaultPage || 0);
  return { page, changePage };
};
```